services/cloudca/network_acl_rule.go

```diff
@@ -0,0 +1,110 @@
+package cloudca
+
+import (
+	"encoding/json"
+
+	"github.com/cloud-ca/go-cloudca/api"
+	"github.com/cloud-ca/go-cloudca/services"
+)
+
+type NetworkAclRule struct {
+	Id           string `json:"id,omitempty"`
+	NetworkAclId string `json:"networkAclId,omitempty"`
+	RuleNumber   string `json:"ruleNumber,omitempty"`
+	Cidr         string `json:"cidr,omitempty"`
+	Action       string `json:"action,omitempty"`
+	Protocol     string `json:"protocol,omitempty"`
+	StartPort    string `json:"startPort,omitempty"`
+	EndPort      string `json:"endPort,omitempty"`
+	IcmpType     string `json:"icmpType,omitempty"`
+	IcmpCode     string `json:"icmpCode,omitempty"`
+	TrafficType  string `json:"trafficType,omitempty"`
+	State        string `json:"state,omitempty"`
+}
+
+type NetworkAclRuleService interface {
+	Get(id string) (*NetworkAclRule, error)
+	List() ([]NetworkAclRule, error)
+	ListByNetworkAclId(networkAclId string) ([]NetworkAclRule, error)
+	ListWithOptions(options map[string]string) ([]NetworkAclRule, error)
+	Create(networkAclRule NetworkAclRule) (*NetworkAclRule, error)
+	Update(id string, networkAclRule NetworkAclRule) (*NetworkAclRule, error)
+	Delete(id string) (bool, error)
+}
+
+type NetworkAclRuleApi struct {
+	entityService services.EntityService
+}
+
+func NewNetworkAclRuleService(apiClient api.ApiClient, serviceCode string, environmentName string) NetworkAclRuleService {
+	return &NetworkAclRuleApi{
+		entityService: services.NewEntityService(apiClient, serviceCode, environmentName, NETWORK_ACL_RULE_ENTITY_TYPE),
+	}
+}
+
+func parseNetworkAclRule(data []byte) *NetworkAclRule {
+	networkAclRule := NetworkAclRule{}
+	json.Unmarshal(data, &networkAclRule)
+	return &networkAclRule
+}
+
+func parseNetworkAclRuleList(data []byte) []NetworkAclRule {
+	aclRules := []NetworkAclRule{}
+	json.Unmarshal(data, &aclRules)
+	return aclRules
+}
+
+//Get network acl rule with the specified id for the current environment
+func (networkAclRuleApi *NetworkAclRuleApi) Get(id string) (*NetworkAclRule, error) {
+	data, err := networkAclRuleApi.entityService.Get(id, map[string]string{})
+	if err != nil {
+		return nil, err
+	}
+	return parseNetworkAclRule(data), nil
+}
+
+func (api NetworkAclRuleApi) ListWithOptions(options map[string]string) ([]NetworkAclRule, error) {
```

Comment:
Isn't `api` usually a pointer? I see it is for some (`Get`, `Create`), but not `List` and `ListWithOptions`